src/components/ExpensifyCashLogo.js

```diff
@@ -3,26 +3,41 @@ import PropTypes from 'prop-types';
 import ProductionLogo from '../../assets/images/expensify-cash.svg';
 import DevLogo from '../../assets/images/expensify-cash-dev.svg';
 import StagingLogo from '../../assets/images/expensify-cash-stg.svg';
-import CONFIG from '../CONFIG';
 import CONST from '../CONST';
+import Environment from '../libs/Environment/index';
 
 const propTypes = {
     width: PropTypes.number.isRequired,
     height: PropTypes.number.isRequired,
 };
 
-const ExpensifyCashLogo = (props) => {
-    switch (CONFIG.EXPENSIFY.ENVIRONMENT) {
-        case CONST.ENVIRONMENT.PRODUCTION:
-            return <ProductionLogo width={props.width} height={props.height} />;
-        case CONST.ENVIRONMENT.STAGING:
-            return <StagingLogo width={props.width} height={props.height} />;
-        default:
-            return <DevLogo width={props.width} height={props.height} />;
+class ExpensifyCashLogo extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            environment: CONST.ENVIRONMENT.PRODUCTION,
+        };
     }
-};
 
-ExpensifyCashLogo.propTypes = propTypes;
-ExpensifyCashLogo.displayName = 'ExpensifyCashLogo';
+    componentDidMount() {
```

Comment:
Since a lot of this code is duplicated between the logo and the badge, I think this could be DRYed up more by moving the state and the component did mount to a parent component. The parent component would then display either the badge or the logo (maybe depending on a `type` prop) and pass the environment down as a prop.

A slight alternative to this would be to use a `withEnvironment` HOC.

Reply:
I like the idea of using an HOC for this more than making a parent component, I'll go that route